### ISSUE TITLE
Latest Windows Installer fails miserably

### ISSUE THREAD

**Issue by djarvis** [codeproject/CodeProject.AI-Server#228]
Clean machine with .NET 9 installed.

Ran **CodeProject.AI-Server_2.9.5_win_x64.exe** and went through the wizard.

No errors appear yet service is not installed and this appears in the event viewer:

Faulting application name: CodeProject.AI.Server.exe, version: 2.7.0.0, time stamp: 0x67200000
Faulting module name: coreclr.dll, version: 9.0.24.52809, time stamp: 0x672049fc
Exception code: 0xc0000602
Fault offset: 0x000000000032a356
Faulting process id: 0x24d8
Faulting application start time: 0x01db4bebe97c51bb
Faulting application path: C:\Program Files\CodeProject\AI\server\CodeProject.AI.Server.exe
Faulting module path: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\9.0.0\coreclr.dll
Report Id: f1bc4a62-e54b-4189-a072-449f334df817
Faulting package full name: 
Faulting package-relative application ID: 

Running setup.bat in C:\program files\CodeProject\AI yields:

             _Installing CodeProject.AI Analysis Module

======================================================================

                   CodeProject.AI Installer

======================================================================

250.3Gb of 953Gb available on (No label) (Windows 10 x86_64 - windows)

General CodeProject.AI setup

Checking for VC++ Redist...v14 Present.
Creating Directories...done

GPU support

CUDA Present...Yes (CUDA 12.3, cuDNN 8.5, CUDA Toolkit: false)
Updating path to include cuDNN
ROCm Present...No
Checking for .NET 9.0...Checking runtimes...All good. Found .NET aspnetcore 9
Cannot find file "C:\Program Files\CodeProject\AI\modules\AI\modulesettings.json"

Reading  settings.......done
Installing module   (Internal)

This module cannot be installed on this system

Setup complete

Total setup time 00:00:01.98_

**Comment by djarvis** [comment]
Version 2.8.0 works fine.  

Some wonky dependency issue with 2.9.5:

Faulting application name: CodeProject.AI.Server.exe, version: 2.7.0.0, time stamp: 0x67200000
Faulting module name: coreclr.dll, version: 9.0.24.52809, time stamp: 0x672049fc
Exception code: 0xc0000602
Fault offset: 0x000000000032a356
Faulting process id: 0x1a0c
Faulting application start time: 0x01db4bef376ad88e
Faulting application path: C:\Program Files\CodeProject\AI\server\CodeProject.AI.Server.exe
Faulting module path: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\9.0.0\coreclr.dll

**Comment by ChrisMaunder** [comment]
What is your: 
 - OS
 - architecture (arm64 or x64)
 - environment (native OS install or using a VM)

**Comment by djarvis** [comment]
Windows Server 2022 Standard
x64: Processor	13th Gen Intel(R) Core(TM) i5-13500, 2500 Mhz, 14 Core(s), 20 Logical Processor(s)
Native OS install.

**Comment by ChrisMaunder** [comment]
This looks like either an oddity with Windows Server vs Windows (consumer) or a permissions issue. We only support (and have only tested on) the consumer versions of windows, not the server version. My only suggestion is to check the event viewer to see if that sheds any light.

**Comment by djarvis** [comment]
Probably not a permissions issue and probably not an OS-specific oddity as all prior versions work just fine.  Maybe some new .NET 9 thing?  I can try to pull down the source and build and run in debug and see what happens.  Event viewer just shows what I pasted above:

Faulting application name: CodeProject.AI.Server.exe, version: 2.7.0.0, time stamp: 0x67200000
Faulting module name: coreclr.dll, version: 9.0.24.52809, time stamp: 0x672049fc
Exception code: 0xc0000602
Fault offset: 0x000000000032a356
Faulting process id: 0x1a0c
Faulting application start time: 0x01db4bef376ad88e
Faulting application path: C:\Program Files\CodeProject\AI\server\CodeProject.AI.Server.exe
Faulting module path: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\9.0.0\coreclr.dll

**Comment by VorlonCD** [comment]
Try uninstalling, deleting `C:\Program Files\CodeProject` before re-installing.

If still failing any thing of interest in here?

```
C:\Program Files\CodeProject\AI\logs\*.txt
C:\Program Files\CodeProject\AI\modules\ObjectDetectionYOLOv5-6.2\install.log
C:\Program Files\CodeProject\AI\modules\ObjectDetectionYOLOv5Net\install.log

```

Maybe uninstall/reinstall .net 9 and reboot?

If you run `dotnet --list-runtimes` at a command prompt, you should see these, and they should all be the same build:

```
Microsoft.AspNetCore.App 9.0.0 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
Microsoft.NETCore.App 9.0.0 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
Microsoft.WindowsDesktop.App 9.0.0 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
```

The last one might not be needed but I always keep all three installed and at the exact same build.

Maybe disable antivirus?

Might get something else out of analyzing this file with WinDbg:
C:\Windows\System32\config\systemprofile\AppData\Local\CrashDumps\CodeProject.AI.Server.exe.*.dmp

**Comment by BuxtonCalvin** [comment]
I had the same installation problem and could not get the service to start.  Using MS Windows Pro 11 latest updates and patches.   

The solution was to delete the old AI sub-directory and all contents.  Then manually install MS net (https://dotnet.microsoft.com/en-us/download/dotnet/thank-you/sdk-9.0.101-windows-x64-installer).  Reinstall code project 2.9.5, and the service then came up in my browser.

**Comment by randellhodges** [comment]
I have installed various versions, up to 2.6.5 on Windows Server 2022 (w/desktop ui) and haven't had any issue. Did you do a core install or install with a desktop? Right clicking and running as administrator? Maybe something broke after 2.6.5?

**Comment by ChrisMaunder** [comment]
@BuxtonCalvin When you installed the latest, did you check the "delete old install and all data"  checkbox?

**Comment by wimmme** [comment]
Hi, just to chime in. I am running CPAI for some years now,  from almost the beginning actually.
It's running on a Windows Server 2022, and the installer almost never worked correctly (especially the modules part).
This one does !  (I did check the "delete old install and all data" option) .Great work !

`Server version:   2.9.5
System:           Windows
Operating System: Windows (Windows 10 Redstone)
CPUs:             11th Gen Intel(R) Core(TM) i7-11700K @ 3.60GHz (Intel)
                  1 CPU x 8 cores. 16 logical processors (x64)
GPU (Primary):    NVIDIA GeForce GT 1030 (2 GiB) (NVIDIA) 
                  Driver: 560.76, CUDA: 12.6.20 (up to: 12.6), Compute: 6.1, cuDNN: 8.6
System RAM:       64 GiB
Platform:         Windows
BuildConfig:      Release
Execution Env:    Native
Runtime Env:      Production
Runtimes installed:
  .NET runtime:     9.0.0
  .NET SDK:         Not found
  Default Python:   3.12.8
  Go:               Not found
  NodeJS:           22.12.0
  Rust:             Not found
Video adapter info:
  NVIDIA GeForce GT 1030:
    Driver Version     32.0.15.6076
    Video Processor    NVIDIA GeForce GT 1030
System GPU info:
  GPU 3D Usage       31%
  GPU RAM Usage      1 GiB
Global Environment variables:
  CPAI_APPROOTPATH = <root>
  CPAI_PORT        = 32168`

**Comment by BuxtonCalvin** [comment]
> @BuxtonCalvin When you installed the latest, did you check the "delete old install and all data" checkbox?

Yes.  I installed with the checkbox checked.  I also deleted the install from the Windows Apps menu.  I went back and forth between installing working versions (2.8.0) and non-working (2.9.5) before applying the fix as per my note above.  In my case, I'm pretty sure this was a dotnet issue.

**Comment by djarvis** [comment]
I have not had the time unfortunately to experiment with the latest version.  I have the 2.8.0 working and need my security camera system up and running

**Comment by RevEngOps** [comment]
I had a similar problem with 2.9.5 while trying to install on Windows 10 Pro.  Looks like .NET 9 defaults to enabling Intel CET which isn't supported on Intel chips older than generation 11 and AMD Ryzen 3+ - AMD has their own shadow-stack protection implementation.  I let the installer finish (it did not install a service), then ran this Powershell command to explicitly disable shadow-stack protection for the server executable:  "Set-ProcessMitigation -Name "CodeProject.AI.Server.exe" -Disable UserShadowStack".  Was finally able to get the server process to run without crashing.  Error when executing prior to disabling shadow stack was referencing this location as the fault:  "File: D:\a\_work\1\s\srce\clreclr\vm\threads.cpp:7938 Image: C:\Program Files\CodeProject\AI\Server\CodeProject.AI.Server.exe".  Of course I still ran into a ton of trouble regarding installing modules, I'm guessing because the 'setup.bat' script never ran to set up the initial environment.  I was able to stumble through getting all of the Python virtual environments setup (and all requirements) and manually downloaded the required models and placed them where they needed to be.  As of right now, it seems that I have a working 'Object Detection (YOLOv5 3.1)' module for my BlueIris install again.  Think the project needs to be re-built with Intel CET disabled.
